server/handler/assignments.go

```diff
@@ -28,6 +28,10 @@ func GetAssignmentsForUserExperiment(repo *repository.Assignments) RequestProces
 		assignments, err := repo.GetAll(userID, experimentID)
 		if err == repository.ErrNoAssignmentsInitialized {
 			if assignments, err = repo.Initialize(userID, experimentID); err != nil {
+				if err == repository.ErrNoPairsAvailable {
```

Comment:
it's incorrect if we are following REST. List APIs should return an empty list when nothing is found, but a request is correct. There is no special rule for it because actually spec for REST doesn't exist. But if you look at any REST API (github for example) you will notice it works like that.

Reply:
404 error for endpoint `/api/:id/list` means there is no such API or id isn't found. Not empty list.

Reply:
`/api/experiments/{experimentId}/assignments` is requesting for the assignments of a given experiment, so this `404 Not Found` means that the collection does not exist at all. It is not an "empty" case.

About the status code: `404 Not Found` it's defined as 
>_The requested resource could not be found but may be available in the future. Subsequent requests by the client are permissible._

what matches our case: if in the future are loaded file-pairs, the endpoint will return data.

Reply:
`/api/experiments/{experimentId}/assignments` is requesting for the assignments of a given experiment, so this `404 Not Found` means that the collection does not exist at all. It is not an "empty" case.

About the status code: `404 Not Found` it's defined as 
>_The requested resource could not be found but may be available in the future. Subsequent requests by the client are permissible._

what matches our case: if in the future are loaded file-pairs, the endpoint will return data.